### ISSUE TITLE
Liftover: NoClassDefFoundError: Could not initialize class __C147RGContainer_GRCh38

### ISSUE THREAD

**Issue by DSuveges** [hail-is/hail#10682]
When trying to lift over the gnomAD 3.1 hail table to GRCh37, I encounter the error mentioned above.

This code works:

```python
import hail as hl

hail_table = 'gs://gcp-public-data--gnomad/release/3.1.1/ht/genomes/gnomad.genomes.v3.1.1.sites.ht'
chain_file = 'gs://hail-common/references/grch38_to_grch37.over.chain.gz'
ht = hl.read_table(hail_table).head(10_000)

GRCh37 = hl.get_reference('GRCh37')
GRCh38 = hl.get_reference('GRCh38')
GRCh38.add_liftover(chain_file, GRCh37)

hl.eval(hl.liftover(hl.locus('chr1', 1034245, 'GRCh38'), 'GRCh37'))
#  Locus(contig=1, position=969625, reference_genome=GRCh37)
```

However, when trying to lift over the entire table it fails:
```
ht = ht.annotate(
    locus_GRCh37 = hl.liftover(ht.locus, 'GRCh37')
)
ht.show()
```

I got the same error when trying to lift over an older gnomAD version (2.1) from GRCh37 to GRCh38, which used to work according to my best knowledge. Also, this way of lifting over a hail table is following the recommended process on the documentation [here](https://hail.is/docs/0.2/guides/genetics.html?highlight=prs#liftover-variants-from-one-coordinate-system-to-another). I'm quite confident there must be something I'm doing wrong, but now I'm stuck, any help would be highly welcome. Thanks!

The code is running on a Google Cloud Dataproc cluster, Python 3.8, hail version: `'0.2.71-f3a54b530979'`

Error stack:
```python
--------------------------------------------------------------------------- / 1]
FatalError                                Traceback (most recent call last)
/opt/conda/miniconda3/lib/python3.8/site-packages/IPython/core/formatters.py in __call__(self, obj)
    700                 type_pprinters=self.type_printers,
    701                 deferred_pprinters=self.deferred_printers)
--> 702             printer.pretty(obj)
    703             printer.flush()
    704             return stream.getvalue()

/opt/conda/miniconda3/lib/python3.8/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    392                         if cls is not object \
    393                                 and callable(cls.__dict__.get('__repr__')):
--> 394                             return _repr_pprint(obj, self, cycle)
    395
    396             return _default_pprint(obj, self, cycle)

/opt/conda/miniconda3/lib/python3.8/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    698     """A pprint that just redirects to the normal repr function."""
    699     # Find newlines and replace them with p.break_()
--> 700     output = repr(obj)
    701     lines = output.splitlines()
    702     with p.group():

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/table.py in __repr__(self)
   1295
   1296         def __repr__(self):
-> 1297             return self.__str__()
   1298
   1299         def data(self):

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/table.py in __str__(self)
   1292
   1293         def __str__(self):
-> 1294             return self._ascii_str()
   1295
   1296         def __repr__(self):

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/table.py in _ascii_str(self)
   1318                 return s
   1319
-> 1320             rows, has_more, dtype = self.data()
   1321             fields = list(dtype)
   1322             trunc_fields = [trunc(f) for f in fields]

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/table.py in data(self)
   1302                 row_dtype = t.row.dtype
   1303                 t = t.select(**{k: hl._showstr(v) for (k, v) in t.row.items()})
-> 1304                 rows, has_more = t._take_n(self.n)
   1305                 self._data = (rows, has_more, row_dtype)
   1306             return self._data

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/table.py in _take_n(self, n)
   1449             has_more = False
   1450         else:
-> 1451             rows = self.take(n + 1)
   1452             has_more = len(rows) > n
   1453             rows = rows[:n]

<decorator-gen-1116> in take(self, n, _localize)

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/typecheck/check.py in wrapper(__original_func, *args, **kwargs)
    575     def wrapper(__original_func, *args, **kwargs):
    576         args_, kwargs_ = check_all(__original_func, args, kwargs, checkers, is_method=is_method)
--> 577         return __original_func(*args_, **kwargs_)
    578
    579     return wrapper

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/table.py in take(self, n, _localize)
   2119         """
   2120
-> 2121         return self.head(n).collect(_localize)
   2122
   2123     @typecheck_method(n=int)

<decorator-gen-1110> in collect(self, _localize)

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/typecheck/check.py in wrapper(__original_func, *args, **kwargs)
    575     def wrapper(__original_func, *args, **kwargs):
    576         args_, kwargs_ = check_all(__original_func, args, kwargs, checkers, is_method=is_method)
--> 577         return __original_func(*args_, **kwargs_)
    578
    579     return wrapper

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/table.py in collect(self, _localize)
   1918         e = construct_expr(rows_ir, hl.tarray(t.row.dtype))
   1919         if _localize:
-> 1920             return Env.backend().execute(e._ir)
   1921         else:
   1922             return e

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/backend/py4j_backend.py in execute(self, ir, timed)
     96                 raise HailUserError(message_and_trace) from None
     97
---> 98             raise e

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/backend/py4j_backend.py in execute(self, ir, timed)
     72         # print(self._hail_package.expr.ir.Pretty.apply(jir, True, -1))
     73         try:
---> 74             result = json.loads(self._jhc.backend().executeJSON(jir))
     75             value = ir.typ._from_json(result['value'])
     76             timings = result['timings']

/usr/lib/spark/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py in __call__(self, *args)
   1302
   1303         answer = self.gateway_client.send_command(command)
-> 1304         return_value = get_return_value(
   1305             answer, self.gateway_client, self.target_id, self.name)
   1306

/opt/conda/miniconda3/lib/python3.8/site-packages/hail/backend/py4j_backend.py in deco(*args, **kwargs)
     28                 raise FatalError('Error summary: %s' % (deepest,), error_id) from None
     29             else:
---> 30                 raise FatalError('%s\n\nJava stack trace:\n%s\n'
     31                                  'Hail version: %s\n'
     32                                  'Error summary: %s' % (deepest, full, hail.__version__, deepest), error_id) from None

FatalError: NoClassDefFoundError: Could not initialize class __C147RGContainer_GRCh38

Java stack trace:
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 20 times, most recent failure: Lost task 0.19 in stage 0.0 (TID 19) (my-first-hail-cluster-w-0.c.open-targets-eu-dev.internal executor 1): java.lang.NoClassDefFoundError: Could not initialize class __C147RGContainer_GRCh38
	at __C144Compiled.applyregion0_8(Emit.scala)
	at __C144Compiled.apply(Emit.scala)
	at is.hail.expr.ir.TableMapRows.$anonfun$execute$43(TableIR.scala:1938)
	at scala.runtime.java8.JFunction1$mcJJ$sp.apply(JFunction1$mcJJ$sp.java:23)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:461)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:496)
	at is.hail.utils.richUtils.RichContextRDD$$anon$1.next(RichContextRDD.scala:79)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:496)
	at is.hail.utils.package$.getIteratorSizeWithMaxN(package.scala:415)
	at is.hail.rvd.RVD.$anonfun$head$2(RVD.scala:526)
	at is.hail.rvd.RVD.$anonfun$head$2$adapted(RVD.scala:526)
	at is.hail.sparkextras.ContextRDD.$anonfun$runJob$2(ContextRDD.scala:366)
	at is.hail.sparkextras.ContextRDD.sparkManagedContext(ContextRDD.scala:164)
	at is.hail.sparkextras.ContextRDD.$anonfun$runJob$1(ContextRDD.scala:365)
	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2242)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2254)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2203)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2202)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2202)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1078)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1078)
	at scala.Option.foreach(Option.scala:407)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1078)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2441)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2383)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2372)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:868)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2202)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2223)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2242)
	at is.hail.sparkextras.ContextRDD.runJob(ContextRDD.scala:362)
	at is.hail.rvd.RVD.$anonfun$head$1(RVD.scala:526)
	at is.hail.utils.PartitionCounts$.incrementalPCSubsetOffset(PartitionCounts.scala:73)
	at is.hail.rvd.RVD.head(RVD.scala:526)
	at is.hail.expr.ir.TableSubset.execute(TableIR.scala:1380)
	at is.hail.expr.ir.TableSubset.execute$(TableIR.scala:1377)
	at is.hail.expr.ir.TableHead.execute(TableIR.scala:1386)
	at is.hail.expr.ir.TableMapRows.execute(TableIR.scala:1905)
	at is.hail.expr.ir.Interpret$.run(Interpret.scala:784)
	at is.hail.expr.ir.Interpret$.alreadyLowered(Interpret.scala:56)
	at is.hail.expr.ir.InterpretNonCompilable$.interpretAndCoerce$1(InterpretNonCompilable.scala:16)
	at is.hail.expr.ir.InterpretNonCompilable$.rewrite$1(InterpretNonCompilable.scala:53)
	at is.hail.expr.ir.InterpretNonCompilable$.$anonfun$apply$1(InterpretNonCompilable.scala:25)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at is.hail.expr.ir.InterpretNonCompilable$.rewriteChildren$1(InterpretNonCompilable.scala:25)
	at is.hail.expr.ir.InterpretNonCompilable$.rewrite$1(InterpretNonCompilable.scala:54)
	at is.hail.expr.ir.InterpretNonCompilable$.apply(InterpretNonCompilable.scala:58)
	at is.hail.expr.ir.lowering.InterpretNonCompilablePass$.transform(LoweringPass.scala:67)
	at is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$3(LoweringPass.scala:15)
	at is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:81)
	at is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$1(LoweringPass.scala:15)
	at is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:81)
	at is.hail.expr.ir.lowering.LoweringPass.apply(LoweringPass.scala:13)
	at is.hail.expr.ir.lowering.LoweringPass.apply$(LoweringPass.scala:12)
	at is.hail.expr.ir.lowering.InterpretNonCompilablePass$.apply(LoweringPass.scala:62)
	at is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1(LoweringPipeline.scala:14)
	at is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1$adapted(LoweringPipeline.scala:12)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
	at is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:12)
	at is.hail.expr.ir.CompileAndEvaluate$._apply(CompileAndEvaluate.scala:29)
	at is.hail.backend.spark.SparkBackend._execute(SparkBackend.scala:381)
	at is.hail.backend.spark.SparkBackend.$anonfun$execute$1(SparkBackend.scala:365)
	at is.hail.expr.ir.ExecuteContext$.$anonfun$scoped$3(ExecuteContext.scala:47)
	at is.hail.utils.package$.using(package.scala:627)
	at is.hail.expr.ir.ExecuteContext$.$anonfun$scoped$2(ExecuteContext.scala:47)
	at is.hail.utils.package$.using(package.scala:627)
	at is.hail.annotations.RegionPool$.scoped(RegionPool.scala:17)
	at is.hail.expr.ir.ExecuteContext$.scoped(ExecuteContext.scala:46)
	at is.hail.backend.spark.SparkBackend.withExecuteContext(SparkBackend.scala:275)
	at is.hail.backend.spark.SparkBackend.execute(SparkBackend.scala:362)
	at is.hail.backend.spark.SparkBackend.$anonfun$executeJSON$1(SparkBackend.scala:406)
	at is.hail.utils.ExecutionTimer$.time(ExecutionTimer.scala:52)
	at is.hail.backend.spark.SparkBackend.executeJSON(SparkBackend.scala:404)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)

java.lang.NoClassDefFoundError: Could not initialize class __C147RGContainer_GRCh38
	at __C144Compiled.applyregion0_8(Emit.scala)
	at __C144Compiled.apply(Emit.scala)
	at is.hail.expr.ir.TableMapRows.$anonfun$execute$43(TableIR.scala:1938)
	at scala.runtime.java8.JFunction1$mcJJ$sp.apply(JFunction1$mcJJ$sp.java:23)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:461)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:496)
	at is.hail.utils.richUtils.RichContextRDD$$anon$1.next(RichContextRDD.scala:79)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:496)
	at is.hail.utils.package$.getIteratorSizeWithMaxN(package.scala:415)
	at is.hail.rvd.RVD.$anonfun$head$2(RVD.scala:526)
	at is.hail.rvd.RVD.$anonfun$head$2$adapted(RVD.scala:526)
	at is.hail.sparkextras.ContextRDD.$anonfun$runJob$2(ContextRDD.scala:366)
	at is.hail.sparkextras.ContextRDD.sparkManagedContext(ContextRDD.scala:164)
	at is.hail.sparkextras.ContextRDD.$anonfun$runJob$1(ContextRDD.scala:365)
	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2242)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)




Hail version: 0.2.71-f3a54b530979
Error summary: NoClassDefFoundError: Could not initialize class __C147RGContainer_GRCh38
```

**Comment by tpoterba** [comment]
Thanks for the detailed error report! This is a known problem, fix is in PR: #10523

**Comment by DSuveges** [comment]
Oh, thank you so much for your quick answer! I'll keep my eyes on the PR!

**Comment by johnc1231** [comment]
@DSuveges , we just released a new version of hail, 0.2.73, which contains this fix.

**Comment by DSuveges** [comment]
Thanks @johnc1231 for letting me know! I have tried the new version and works like a charm. Great work!
